### PR TITLE
use common task ID instead of map key

### LINKED ISSUE #158886: Reload window creates new task
Testing #158787

- open vscode workspace in VS Code
- F1 -> "Run Task" -> "Run Code Server"

Observe: a task show up:
CleanShot 2022-08-23 at 14 09 54@2x

- F1 -> "Reload Window"

Observe: another "Run Code Server" shows up:
CleanShot 2022-08-23 at 14 11 05@2x

### PR DESCRIPTION
**Pull request by meganrogge**
fix #158886

works in oss for all tasks - making a build to double check

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/test/browser/links/terminalLocalLinkDetector.test.ts [hunk 1]
```diff
@@ -144,7 +145,7 @@ suite('Workbench - TerminalLocalLinkDetector', () => {
 		});
 	});
 
-	suite('Windows', () => {
+	suite.only('Windows', () => {
```
Comment (mjbvz) [inline src/vs/workbench/contrib/terminal/test/browser/links/terminalLocalLinkDetector.test.ts:148]: Looks like a missed `.only`

**Review by mjbvz — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/test/browser/links/terminalLocalLinkDetector.test.ts (hunk 1 above)
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/test/browser/links/terminalLocalLinkDetector.test.ts:148]:
```suggestion
	suite('Windows', () => {
```

**Review by mjbvz — DISMISSED** [review]

**Comment by meganrogge** [comment]
it works in the build 👍🏼

File: src/vs/workbench/contrib/terminal/test/browser/links/terminalLocalLinkDetector.test.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ const windowsLinks = [
 	'foo/bar',
 	'foo\\bar',
 	'foo\\bar+more',
+	'foo bar'
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/test/browser/links/terminalLocalLinkDetector.test.ts:47]:
```suggestion
```

**Review by DonJayamanne — APPROVED** [review]

**Comment by meganrogge** [comment]
works well - anything I should be concerned abt with this change @alexr00 ?

**Comment by alexr00** [comment]
Looks like it should be safe!

**Pull request merged**